syncstorage/views/__init__.py

```diff
@@ -319,6 +342,84 @@ def post_collection(request):
     return res
 
 
+def post_collection_transaction(request):
+    storage = request.validated["storage"]
+    userid = request.validated["userid"]
+    collection = request.validated["collection"]
+    bsos = request.validated["bsos"]
+    invalid_bsos = request.validated["invalid_bsos"]
+    batch = request.validated["batch"]
+    commit = request.validated["commit"]
+
+    # The "batch" key is set only on a multi-POST batch request prior to a
+    # commit.  The "modified" key is only set upon a successful commit.
+    # The two flags are mutually exclusive.
+    # Any failures at all mean cancelling the transaction completely.
+    res = {'success': [], 'failed': {}}
+
+    # If there are any parsing failures, we won't even start a transaction.
```

Comment:
We should check this behaviour with @rnewman, since what you describe differs from behaviour in the non-transactional case.  In that case the valid bsos are written and the invalid ones are rejected, and I can see an argument that we should do the same here, report the failures to the client, and let it decide what to do about the problem.